src/items/enumerations.md

```diff
@@ -122,6 +155,59 @@ enum OverflowingDiscriminantError2 {
 }
 ```
 
+### Accessing Discriminant
+
+#### Via `mem::discriminant`
+
+[`mem::discriminant`] returns an opaque reference to the discriminant of
+an enum value which can be compared. This cannot be used to get the value
+of the discriminant.
+
+#### Casting
+
+If an enumeration is C-like (with no tuple and struct variants), then its
+discriminant can be directly accessed with a [numeric cast]; e.g.:
```

Comment:
This seems to be missing the grandfathered support for casting enums with empty constructors, like:

```rust
enum Fieldless {
    Tuple(),
    Struct{},
    Unit,
}

assert_eq!(Fieldless::Tuple() as u8, 0);
assert_eq!(Fieldless::Struct{} as u8, 1);
assert_eq!(Fieldless::Unit as u8, 2);
```

Was that excluded for a reason?

Reply:
The problem is that they cannot be casted to their discriminant if their discriminants are specified. This is summarized in https://github.com/rust-lang/rust/issues/60553#issuecomment-985007785. This will likely need more discussion around the design, but none of the design concerns block the stabilization of arbitrary_enum_discriminant.

If you would like this to be included, then I can put that in.